ts/state/ducks/search.ts

```diff
@@ -98,12 +98,35 @@ async function doSearch(
 ): Promise<SearchResultsPayloadType> {
   const { regionCode } = options;
 
+  const advancedSearchOptions = getAdvancedSearchOptionsFromQuery(query);
+  const processedQuery = advancedSearchOptions.query;
+  const isAdvancedQuery = query !== processedQuery;
+
   const [discussions, messages] = await Promise.all([
-    queryConversationsAndContacts(query, options),
-    queryMessages(query),
+    queryConversationsAndContacts(processedQuery, options),
+    queryMessages(processedQuery),
   ]);
   const { conversations, contacts } = discussions;
-  const filteredMessages = messages.filter(message => message !== undefined);
+  let filteredMessages = messages.filter(message => message !== undefined);
+
+  if (isAdvancedQuery) {
+    let senderFilter: Array<string> = [];
+    if (
+      advancedSearchOptions.from !== null &&
```

Comment:
```suggestion
      advancedSearchOptions.from &&
```
We can do this boolean check here instead. It will return `true` if `from` is not an invalid value (`null`, `undefined`, `false`, `0`)

Reply:
Will be included in the next commit, thanks